src/cudamatrix/cu-matrix.cc

```diff
@@ -1489,36 +1489,13 @@ void CuMatrixBase<Real>::FindRowMaxId(CuArray<int32> *id) const {
 #if HAVE_CUDA == 1
   if (CuDevice::Instantiate().Enabled()) {
     Timer tim;
-
-    // initialize the vectors
-    CuVector<Real> max(num_rows_);
-    max.Set(-1e21);
     id->Resize(num_rows_);
-    id->Set(-1);
+    MatrixDim d = Dim();
+    dim3 dimBlock(CU1DBLOCK);
```

Comment:
Here we could add a comment describing the special layout of CUDA blocks and grids:
// CUDA thread layout: one block of threads per matrix-line,